android/src/main/kotlin/ru/innim/my_target_flutter/MethodCallHandler.kt

```diff
@@ -0,0 +1,129 @@
+
+import android.content.Context
+import com.my.target.common.MyTargetConfig
+import com.my.target.common.MyTargetManager
+import io.flutter.plugin.common.MethodCall
+import io.flutter.plugin.common.MethodChannel
+import java.util.*
+import com.my.target.ads.InterstitialAd as InterstitialAd
+
+
+class MethodCallHandler(
+    private val context: Context,
+    private val adEventHandler: AdEventHandler
+) : MethodChannel.MethodCallHandler {
+    companion object {
+        private const val METHOD_INITIAL = "initial"
+        private const val METHOD_CREATE_INTERSTITIAL_AD = "createInterstitialAd"
+        private const val METHOD_LOAD = "load"
+        private const val METHOD_SHOW = "show"
+        private const val INVALID_ARGS_ERROR_CODE = "INVALID_ARGS"
+        private const val NOT_FOUND_ERROR_CODE = "ADS_NOT_FOUND"
+        private const val INVALID_ARGS_ERROR_MESSAGE = "value cant be null"
+
+    }
+
+    private val ads = mutableListOf<AdWithId>()
+
+    override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
+        when (call.method) {
+            METHOD_INITIAL -> {
+                val data = InitialData.fromArgs(call)
+                initAd(context, data.useDebugMode, data.testDevices, result)
+            }
+            METHOD_CREATE_INTERSTITIAL_AD -> {
+                val slotId = call.argument<Int>("slotId")?.toInt()
+                createInterstitialAd(slotId, result)
+            }
+            METHOD_LOAD -> {
+                val id = call.arguments.toString()
+                load(result, id)
+            }
+            METHOD_SHOW -> {
+                val id = call.arguments.toString()
+                show(result, id)
+            }
+        }
+
+    }
+
+    private fun initAd(
+        context: Context,
+        useDebugMode: Boolean,
+        testDevices: String?,
+        result: MethodChannel.Result,
+    ) {
+        MyTargetManager.initSdk(context)
+        MyTargetManager.setDebugMode(useDebugMode)
+        if (testDevices != null) {
+            val configBuilder = MyTargetConfig.Builder().withTestDevices(testDevices).build()
+            MyTargetManager.setSdkConfig(configBuilder)
+        }
+        result.success(true)
+    }
+
+    private fun createInterstitialAd(slotId: Int?, result: MethodChannel.Result) {
+        if (slotId == null) {
+            result.error(INVALID_ARGS_ERROR_CODE, INVALID_ARGS_ERROR_MESSAGE, "slotId: $slotId")
+        } else {
+            val interstitialAd = InterstitialAd(slotId, context)
+            val random = UUID.randomUUID()
+            val id = "${slotId}_${random}"
+            val listener = AdListener(adEventHandler, id) { id: String -> clear(id) }
+            interstitialAd.setListener(listener)
+            ads.add(AdWithId(interstitialAd, id))
+            result.success(id)
+        }
+    }
+
+    private fun load(result: MethodChannel.Result, id: String?) {
+        findAdById(result, id) { ad: InterstitialAd -> ad.load() }
```

Comment:
аргумент, тем более с типом, обычно не пишут, можно просто
```suggestion
        findAdById(result, id) { it.load() }
```